src/yara.cc

```diff
@@ -952,19 +953,118 @@ int scanCallback(int message, void* data, void* param) {
 	return CALLBACK_CONTINUE;
 }
 
+NAN_METHOD(ScannerWrap::ReconfigureRules) {
+	Nan::HandleScope scope;
+
+	ScannerWrap* scanner = ScannerWrap::Unwrap<ScannerWrap>(info.This());
+
+	scanner->lock_read();
+
+	bool rules_compiled = scanner->rules ? true : false;
+
+	if (! rules_compiled) {
+		Nan::ThrowError("Please call configure() before reconfigureRules()");
+		return;
+	}
+	Local<Object> options = Nan::To<Object>(info[0]).ToLocalChecked();
+
+	Local<Array> variables = Local<Array>::Cast(
```

Comment:
This is getting into Node internals that I don't know, so I'm assuming that all these `Local<x>` objects and `Nan::New`/`.toLocalChecked()` calls will have the results correctly freed after `ReconfigureRules` is done, i.e. there's no memory leak.

Reply:
According to the documentation that I've read that is the case as we're using a [normal local handle scope](https://v8.dev/docs/embed#handles-and-garbage-collection).